agent/bench-scripts/templates/make-fio-jobfile.py

```diff
@@ -51,7 +51,7 @@ def replace_val(dct, magic, delta):
     keys = list(dct.keys())
     for k in keys:
         if isinstance(dct[k], dict):
-            dct[k] = replace_val(dct[k], delta)
+            dct[k] = replace_val(dct[k], magic, delta)
         else:
             if (dct[k] is not None) and magic in dct[k]:
                 dct[k] = dct[k].replace(magic, delta[k])
```

Comment:
So the value of `dct[k]` is either a `dict` or a `list`?? It'd be more clear to have an `elif isinstance(dct[k], list)` and maybe an error `else` to be sure; but granting that's a hard convention unlikely to change (?), a comment might be good to explain precisely why we know/assume this.

Reply:
> So the value of `dct[k]` is either a `dict` or a `list`??

Not sure this is true.  You can have a `dict` or a `string` and the `in` operator will return `True` or `False`:

```
[pportant@gandalf pbench]$ python3
Python 3.8.6 (default, Sep 25 2020, 00:00:00)
[GCC 10.2.1 20200723 (Red Hat 10.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> foo = "is this in that string"
>>> "this" in foo
True
>>>
```

> It'd be more clear to have an `elif isinstance(dct[k], list)` and maybe an error `else` to be sure; but granting that's a hard convention unlikely to change (?), a comment might be good to explain precisely why we know/assume this.

I have added assertions to both methods to "document" the expected behavior.  And I verified that assertions do not fire via the existing unit tests we have (adding a `not` to the assertion causes the tests to fail).